server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -1559,10 +1621,25 @@ protected TSStatus executeNonQueryPlan(PhysicalPlan plan) {
         : RpcUtils.getStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR);
   }
 
-  private TSStatus checkPathValidity(String path) {
-    if (!PATH_PATTERN.matcher(path).matches()) {
-      logger.warn("Illegal path: {}", path);
-      return RpcUtils.getStatus(TSStatusCode.PATH_ILLEGAL, path + " path is illegal");
+  private TSStatus checkPathValidityWithNodes(List<String> nodes) {
+    if (nodes == null) {
+      return RpcUtils.getStatus(TSStatusCode.PATH_ILLEGAL, "path is illegal");
+    }
+    int index = 0;
+    if (nodes.get(index).equals(IoTDBConstant.PATH_ROOT)) {
+      for (index = 1; index < nodes.size(); index++) {
+        if (!NODE_NAME_PATTERN.matcher(nodes.get(index)).matches()) {
+          // support double quote on measurement only now
+          if (index == nodes.size() - 1 && NODE_NAME_WITH_QUOTATION_PATTERN.matcher(nodes.get(index)).matches()) {
+            continue;
+          }
```

Comment:
"// support double quote on measurement only now" is not consistent with your PR description.

Reply:
fixed